app/lib/search/mem_index.dart

```diff
@@ -474,23 +478,26 @@ class PackageNameIndex {
 
   /// Search using the parsed [word] and return the match packages with scores.
   Score searchWord(String word, {Set<String>? packages}) {
-    final pkgNamesToCheck = packages ?? _namesWithoutGaps.keys;
+    final pkgNamesToCheck = packages ?? _data.keys;
     final values = <String, double>{};
     final singularWord = word.length <= 3 || !word.endsWith('s')
         ? word
         : word.substring(0, word.length - 1);
-    final parts = singularWord.length <= 3
-        ? [singularWord]
-        : ngrams(singularWord, 3, 3).toList();
+    final collapsedWord = _collapseName(singularWord);
+    final parts =
+        collapsedWord.length <= 3 ? [collapsedWord] : trigrams(collapsedWord);
     for (final pkg in pkgNamesToCheck) {
-      final withoutGaps = _namesWithoutGaps[pkg] ?? pkg;
-      if (withoutGaps.contains(singularWord)) {
+      final entry = _data[pkg];
+      if (entry == null) {
+        continue;
+      }
+      if (entry.collapsed.contains(collapsedWord)) {
         values[pkg] = 1.0;
         continue;
       }
       var matched = 0;
       for (final part in parts) {
-        if (withoutGaps.contains(part)) {
+        if (entry.trigrams.contains(part)) {
```

Comment:
Should we consider set intersection?

Reply:
also, the sane thing is probably an inverse index.

Reply:
Set intersection produced worse results.
inverse index may work, but yeah, live updates are not in favor of it